lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/AbstractTestS3FileSystem.java

```diff
@@ -134,19 +146,30 @@ void testFileWithTrailingWhitespaceAgainstNativeClient()
         try (S3Client s3Client = createS3Client()) {
             String key = "foo/bar with whitespace ";
             byte[] contents = "abc foo bar".getBytes(UTF_8);
-            s3Client.putObject(
-                    request -> request.bucket(bucket()).key(key),
-                    RequestBody.fromBytes(contents.clone()));
+
+            PutObjectRequest putObjectRequest = PutObjectRequest.builder()
```

Comment:
Curiosity question: Can we have for new table formats (Iceberg, Delta Lake) situations where the metadata files are not encrypted, but the content files would be encrypted?

Probably not because the metadata files contain stats which effectively contain content which should be encrypted. I just want to make sure that have this unlikely scenario on the radar.

Reply:
This is outside of the scope of this PR, I think that parquet modular encryption has this scheme.